server/src/main/java/org/elasticsearch/index/IndexSortConfig.java

```diff
@@ -127,11 +129,21 @@ private static MultiValueMode parseMultiValueMode(String value) {
     final FieldSortSpec[] sortSpecs;
     private final Version indexCreatedVersion;
     private final String indexName;
+    private final IndexMode indexMode;
 
     public IndexSortConfig(IndexSettings indexSettings) {
         final Settings settings = indexSettings.getSettings();
         this.indexCreatedVersion = indexSettings.getIndexVersionCreated();
         this.indexName = indexSettings.getIndex().getName();
+        this.indexMode = indexSettings.getMode();
+
+        if (this.indexMode == IndexMode.TIME_SERIES) {
+            this.sortSpecs = new FieldSortSpec[] {
+                new FieldSortSpec(TimeSeriesIdFieldMapper.NAME),
+                new FieldSortSpec(DataStreamTimestampFieldMapper.DEFAULT_PATH) };
+            return;
+        }
```

Comment:
This could probably use the same "move it into a method on IndexMode" treatment that the builder used. But that can wait for a follow up I think.